services/src/main/java/org/keycloak/authentication/requiredactions/DeleteAccount.java

```diff
@@ -107,7 +106,6 @@ public void processAction(RequiredActionContext context) {
         eventBuilder.event(EventType.DELETE_ACCOUNT)
             .client(keycloakContext.getClient())
             .user(user)
-            .detail(Details.USERNAME, user.getUsername())
```

Comment:
Removing this line is fine, as here the username is picked from the user, identical to what we do when constructing the event when passing in the user.